lib/testing/testDataCheck.hpp

```diff
@@ -170,9 +173,11 @@ void testDataCheck<A_Type>::main_thread() {
         frames++;
 
         if (num_frames_to_test == frames) {
-            INFO("Test passed, exiting!");
-            TEST_PASSED();
-        }
+            if (!error) {
```

Comment:
We should rethink this logic.   I like the idea of having it print the first `max_num_errors`, but we shouldn't treat `max_num_errors` as a "maximum acceptable number of errors", and exit for any number of errors in all cases (after printing the first max_num_errors).  Perhaps we should also rename `max_num_errors` to something like `num_errors_logged`?    On that note, the `error` boolean can likely be replaced with just `num_errors > 0`.